.circleci/config.yml

```diff
@@ -172,19 +172,65 @@ jobs:
             cd /tmp/keep-client/contracts
             gsutil -m cp * gs://${CONTRACT_DATA_BUCKET}/keep-core
 
-  build_token_dashboard_dapp:
+  publish_npm_package:
     executor: docker-node
     steps:
       - attach_workspace:
           at: /tmp/keep-client
-      - setup_remote_docker:
-          docker_layer_caching: true
       - checkout
       - run:
           name: Copy contract artifacts
-          working_directory: ~/project/solidity/dashboard
+          working_directory: ~/project/solidity
           command: |
-            ./scripts/copy-contracts.sh /tmp/keep-client/contracts
+            set -x
+            mkdir -p artifacts
+            cp -r /tmp/keep-client/contracts/* artifacts/
+            name=$(jq --raw-output .name package.json)
+            version=$(jq --raw-output .version package.json)
+            preid=$(echo $version | sed -e s/^.*-\\\([^.]*\\\).*$/\\1/)
+
+            # Find the latest published package version matching this preid.
+            latest_version=$(npm show -json "$name@^$version" version | jq --raw-output ". | sort | .[-1]")
+            latest_version=${latest_version:-$version}
+            if [ -z $latest_version ]; then
+              echo "Latest version calculation failed. Resolved info:"
+              echo "$name@$version ; preid $preid"
+              exit 1
+            fi
+
+            # Update package.json with the latest published package version matching this
+            # preid to prepare for bumping.
+            echo $(jq -M ".version=\"${latest_version}\"" package.json) > package.json
+
+            # Bump without doing any git work. Versioning is a build-time action for us.
+            # Consider including commit id? Would be +<commit id>.
+            npm version prerelease --preid=$preid --no-git-tag-version
+
+            # Fix resolved dependency versions.
+            npm update
+
+            # Publish to npm.
+            echo //registry.npmjs.org/:_authToken=${NPM_TOKEN} > .npmrc
```

Comment:
We're storing `NPM_TOKEN` in a Circle Context?  Will this token be the same across tbtc and keep-ecdsa?  If we require different tokens and this is in a Circle context we'll need to supplement the token name.

Reply:
The token should be the same across repos, yeah. It's for a `thesis-heimdall` account on npm, at least for now.

I don't love that `thesis-heimdall` can deploy with the token even if we're releasing a final version, but… Need to cross that bridge separately.